packages/graphql/src/schema-model/parser/parse-annotation.ts

```diff
@@ -52,66 +28,14 @@ export function parseAnnotations(directives: readonly DirectiveNode[]): Annotati
             // multiple interfaces can have this annotation - must constrain this flexibility by design
             return directivesMap;
         }
-        const annotation = parseDirective(directive);
+        const annotation = annotationsParsers[directive.name.value]?.(
+            directive,
+            directives.filter((other) => other.name.value === directive.name.value)
+        );
```

Comment:
This bit here is quite hard to follow, it would be worth adding some intermediate variables to aid readability

However, more importantly:
* I've checked the typings, the result of `annotationsParsers[directive.name.value]` is any, removing any form of static type checking on this bit here, which includes the annotation that gets returned by it
* I've noticed that here we are passing 2 parameters to the parsing function, however, most parsing functions only accept one parameter (I think key is the exception). This bit here is relying on the lack of typings of `annotationsParsers[directive.name.value]` to handle the different interfaces implicitly. This is extremely brittle, as any annotationParser adding a different interface (which may happen and is the reasoning behind having different parsers) will silently fail